Assignment6.Rmd

```diff
@@ -70,14 +70,34 @@ table(M2$certified, M2$predict1)
 
 table(M2$certified, M2$predict2)
 
+accuracy1 <- sum(M2$predict1 == M2$certified)/length(M2$predict1)
+accuracy1
+
+accuracy2 <- sum(M2$predict2 == M2$certified)/length(M2$predict2)
+accuracy2
+
+#tree1 has an accuracy rate of 0.2186; tree2 has an accuracy rate of 0.5363 so tree2 has a lower error rate
 ```
 
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
+library(readr)
+student.course <- read.csv(file = "student.course.csv", header = TRUE)
+
+subject.tree <- rpart(SUBJECT ~ GRD_PTS_PER_UNIT + DIV, method="class", data=student.course)
+printcp(subject.tree)
+post(subject.tree, file = "subject.tree.ps") 
+
+install.packages("PCAmixdata")
+library("PCAmixdata")
+
+X.quanti <- splitmix(student.course)$X.quanti
+X.quali <- splitmix(student.course)$X.quali
 
+pca<-PCAmix(X.quanti,X.quali)
```

Comment:
Good work. You could continue to change a different set of predictors and compare the two models. Using PCA is a good try. Continue to finish that!